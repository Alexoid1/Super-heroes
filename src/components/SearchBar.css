.filter {
  text-align: center;
}

.formSe {
  width: 80%;
  text-align: center;
  margin: auto;
}

.inputSearch input {
  margin: 50px;
  width: 600px;
  height: 30px;
  padding: 7px;
  font-size: 17px;
  font-weight: bolder;
  font-family: 'Chakra Petch', sans-serif;
  background: none;
  color: rgb(255, 196, 0);
  border: none;
  cursor: pointer;
  outline: none;
}

.inputSearch input::placeholder{
  color: rgb(255, 166, 0);
  font-size: 17px;
  font-weight: bolder;
}

.underline {
  color: inherit;
  text-decoration: none;
}

.underline {
  background: 
    linear-gradient(to right, rgba(100, 200, 200, 1), rgba(100, 200, 200, 1)),
    linear-gradient(to right, rgba(255, 0, 0, 1), rgba(255, 0, 180, 1), rgba(0, 100, 200, 1));
  background-size: 100% 0.1em, 0 0.1em;
  background-position: 100% 100%, 0 100%;
  background-repeat: no-repeat;
  transition: background-size 400ms;
}

.underline:hover,
.underline:focus {
  background-size: 0 0.1em, 100% 0.1em;
}

@media (max-width: 880px) {
  .inputSearch input {
    width: 300px;
  }
}

@media (max-width: 420px) {
  .inputSearch input {
    width: 200px;
    margin: 5px;
  }
}
