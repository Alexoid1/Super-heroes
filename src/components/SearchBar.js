import React, { useState } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { changeText } from '../actions/index';
import './SearchBar.css';

const SearchBar = ({ heroes, changeText, onChange }) => {
  const [text, setText] = useState(heroes.text);
  const handleTextChange = (event) => {
    setText(event.target.value);
    onChange(event.target.value);
    changeText(event.target.value);
  };

  return (
    <>
      <div>
        <form className="formSe">
          <div className="inputSearch">
            <input
              type="text"
              value={text}
              onChange={handleTextChange}
              placeholder="Hero Name"
            />
          </div>

        </form>
      </div>
    </>
  );
};

SearchBar.propTypes = {
  heroes: PropTypes.shape({
    loading: PropTypes.bool.isRequired,
    heroes: PropTypes.arrayOf(PropTypes.object),
    sHeroes: PropTypes.arrayOf(PropTypes.object),
    error: PropTypes.string.isRequired,
    startIndex: PropTypes.number.isRequired,
    lastIndex: PropTypes.number.isRequired,
    text: PropTypes.string.isRequired,
  }),
  changeText: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
};

SearchBar.defaultProps = {
  heroes: [],
};
const mapStateToProps = (state) => ({
  heroes: state.heroes,
});

const mapDispatchToProps = (dispatch) => ({
  changeText: (text) => dispatch(changeText(text)),
});

export default connect(mapStateToProps, mapDispatchToProps)(SearchBar);
