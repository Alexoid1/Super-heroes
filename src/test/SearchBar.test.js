import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import ShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';
import store from '../reducers/index';
import SearchBar from '../components/SearchBar';

const renderer = new ShallowRenderer();

it('should render SearchBar correctly', () => {
  renderer.render(
    <Provider store={store}>
      <SearchBar />
    </Provider>,
  );
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('find component by placeholder', () => {
  const { getByPlaceholderText } = render(
    <Provider store={store}>
      <SearchBar />
    </Provider>,
  );
  expect(getByPlaceholderText('Hero Name')).toBeInTheDocument();
});
