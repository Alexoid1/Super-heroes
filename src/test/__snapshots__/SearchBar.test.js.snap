// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SearchBar correctly 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Memo(Connect(SearchBar)) />
</Context.Provider>
`;
